src/redux/reservations/reservations.js

```diff
@@ -0,0 +1,84 @@
+const API_RESERVATIONS_INDEX_ENDPOINT = `${process.env.REACT_APP_API_HOST}/api/v1/reservations`;
+const REQUEST_STARTED = 'book-a-lawyer/reservations/REQUEST_STARTED';
+const REQUEST_FAILED = 'book-a-lawyer/reservations/REQUEST_FAILED';
+const REQUEST_COMPLETED = 'book-a-lawyer/reservations/REQUEST_COMPLETED';
+const CLEAR_FETCH_STATE = 'book-a-lawyer/reservationsDelete/CLEAR_FETCH_STATE';
+
+const initialState = {
+  reservations: [],
+  status: 'idle',
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case REQUEST_STARTED:
+      return {
+        ...state,
+        ...action.payload,
+      };
+    case REQUEST_FAILED:
+      return {
+        ...state,
+        ...action.payload,
+      };
+    case REQUEST_COMPLETED:
+      return {
+        ...state,
+        ...action.payload,
+      };
+    case CLEAR_FETCH_STATE:
+      return action.payload;
+    default:
+      return state;
+  }
+};
+
+const requestStarted = () => ({
+  type: REQUEST_STARTED,
+  payload: {
+    status: 'fetching',
+  },
+});
+
+const requestFailed = (error) => ({
+  type: REQUEST_FAILED,
+  payload: {
+    status: 'failed',
+    error,
+  },
+});
+
+const requestCompleted = ({ reservations }) => ({
+  type: REQUEST_COMPLETED,
+  payload: {
+    status: 'completed',
+    reservations,
+  },
+});
+
+export const clearFetchReservationState = () => ({
+  type: CLEAR_FETCH_STATE,
+  payload: {
+    status: 'idle',
+  },
+});
```

Comment:
In the specific case of fetching the reservation_index, we want to keep the list of fetched reservations and it should persist on the state through the session lifetime, having an action for clearing the state in this reducer may cause unnecessary re-fetching the reservations resource every time a component using this reducer is mounted.

```suggestion
```